### PR TITLE
health: add health command

### PR DESCRIPTION
**Pull request by subhamkrai**

### CODE REVIEW

**Comment by travisn** [comment]
> @travisn `odf get cluster-health` sounds good or `odf health` small works better?

Looking back at the ODF CLI design doc, let's go with `odf get health`. It's good to have a verb in the command. With `odf health` there is no verb.

**Comment by subhamkrai** [comment]
based on the PR #23

**Comment by subhamkrai** [comment]
but testing looks good
```yaml
 ./bin/odf get health
Info: Checking if at least three mon pods are running on different nodes
rook-ceph-mon-a-89ddf6c75-sp56z	Running	openshift-storage	ip-10-0-71-84.ec2.internal
rook-ceph-mon-b-9d98dccbc-jb4t7	Running	openshift-storage	ip-10-0-8-11.ec2.internal
rook-ceph-mon-c-5bb58495d5-9tf4c	Running	openshift-storage	ip-10-0-122-244.ec2.internal

Info: Checking mon quorum and ceph health details
Error: HEALTH_ERR

Info: Checking if at least three osd pods are running on different nodes
rook-ceph-osd-0-56b8d6ccc9-mhmls	Running	openshift-storage	ip-10-0-8-11.ec2.internal
rook-ceph-osd-1-7b64c8ff49-njddt	Running	openshift-storage	ip-10-0-71-84.ec2.internal
rook-ceph-osd-2-74557869c6-mqr4n	Running	openshift-storage	ip-10-0-122-244.ec2.internal

Info: Pods that are in 'Running' or `Succeeded` status
csi-addons-controller-manager-b57db8956-dhnhv 	 Running 	 openshift-storage	 ip-10-0-122-244.ec2.internal
csi-cephfsplugin-df97f 	 Running 	 openshift-storage	 ip-10-0-8-11.ec2.internal
csi-cephfsplugin-hqkj4 	 Running 	 openshift-storage	 ip-10-0-71-84.ec2.internal
csi-cephfsplugin-jqpvh 	 Running 	 openshift-storage	 ip-10-0-122-244.ec2.internal
csi-cephfsplugin-provisioner-7468767c47-grxk7 	 Running 	 openshift-storage	 ip-10-0-71-84.ec2.internal
csi-cephfsplugin-provisioner-7468767c47-pwvgc 	 Running 	 openshift-storage	 ip-10-0-8-11.ec2.internal
csi-rbdplugin-5djqv 	 Running 	 openshift-storage	 ip-10-0-8-11.ec2.internal
csi-rbdplugin-bd5cx 	 Running 	 openshift-storage	 ip-10-0-122-244.ec2.internal
csi-rbdplugin-provisioner-7cfc8c7cf7-dl8nv 	 Running 	 openshift-storage	 ip-10-0-8-11.ec2.internal
csi-rbdplugin-provisioner-7cfc8c7cf7-s7pkw 	 Running 	 openshift-storage	 ip-10-0-122-244.ec2.internal
csi-rbdplugin-qzzcv 	 Running 	 openshift-storage	 ip-10-0-71-84.ec2.internal
noobaa-operator-5bbb9cfd85-dnxcd 	 Running 	 openshift-storage	 ip-10-0-122-244.ec2.internal
ocs-metrics-exporter-6c75f7bdc6-cj6f2 	 Running 	 openshift-storage	 ip-10-0-122-244.ec2.internal
ocs-operator-5fcb67864b-vc67t 	 Running 	 openshift-storage	 ip-10-0-122-244.ec2.internal
odf-console-7585b78b4-9frst 	 Running 	 openshift-storage	 ip-10-0-122-244.ec2.internal
odf-operator-controller-manager-78f47ddf65-stpnd 	 Running 	 openshift-storage	 ip-10-0-122-244.ec2.internal
rook-ceph-crashcollector-ip-10-0-122-244.ec2.internal-75f9ppzll 	 Running 	 openshift-storage	 ip-10-0-122-244.ec2.internal
rook-ceph-crashcollector-ip-10-0-71-84.ec2.internal-d76b95fjzrm 	 Running 	 openshift-storage	 ip-10-0-71-84.ec2.internal
rook-ceph-crashcollector-ip-10-0-8-11.ec2.internal-dc84fb7h77qv 	 Running 	 openshift-storage	 ip-10-0-8-11.ec2.internal
rook-ceph-exporter-ip-10-0-122-244.ec2.internal-54d789974-4p7sz 	 Running 	 openshift-storage	 ip-10-0-122-244.ec2.internal
rook-ceph-exporter-ip-10-0-71-84.ec2.internal-7b9b6d7579-9k7jt 	 Running 	 openshift-storage	 ip-10-0-71-84.ec2.internal
rook-ceph-exporter-ip-10-0-8-11.ec2.internal-66f48c8ff9-nwdph 	 Running 	 openshift-storage	 ip-10-0-8-11.ec2.internal
rook-ceph-mgr-a-7dc89565f5-vj879 	 Running 	 openshift-storage	 ip-10-0-71-84.ec2.internal
rook-ceph-mon-a-89ddf6c75-sp56z 	 Running 	 openshift-storage	 ip-10-0-71-84.ec2.internal
rook-ceph-mon-b-9d98dccbc-jb4t7 	 Running 	 openshift-storage	 ip-10-0-8-11.ec2.internal
rook-ceph-mon-c-5bb58495d5-9tf4c 	 Running 	 openshift-storage	 ip-10-0-122-244.ec2.internal
rook-ceph-operator-6557f9c6b5-z2zzv 	 Running 	 openshift-storage	 ip-10-0-8-11.ec2.internal
rook-ceph-osd-0-56b8d6ccc9-mhmls 	 Running 	 openshift-storage	 ip-10-0-8-11.ec2.internal
rook-ceph-osd-1-7b64c8ff49-njddt 	 Running 	 openshift-storage	 ip-10-0-71-84.ec2.internal
rook-ceph-osd-2-74557869c6-mqr4n 	 Running 	 openshift-storage	 ip-10-0-122-244.ec2.internal
rook-ceph-osd-prepare-0e5eaa6f75eba6b7674319908384edd1-rtvg8 	 Succeeded 	 openshift-storage	 ip-10-0-71-84.ec2.internal
rook-ceph-osd-prepare-26463b8fe05e882dc71f99d7ee2f4139-hv8rl 	 Succeeded 	 openshift-storage	 ip-10-0-122-244.ec2.internal
rook-ceph-osd-prepare-9c1e9a35577e54d37b0e0e0f11027626-fxfbv 	 Succeeded 	 openshift-storage	 ip-10-0-8-11.ec2.internal

Warning: Pods that are 'Not' in 'Running' status
noobaa-core-0 	 Pending 	 openshift-storage 	 
noobaa-db-pg-0 	 Pending 	 openshift-storage 	 
rook-ceph-mds-ocs-storagecluster-cephfilesystem-a-84c7df65fsspp 	 Pending 	 openshift-storage 	 
rook-ceph-mds-ocs-storagecluster-cephfilesystem-b-cffbf4bd8bzqw 	 Pending 	 openshift-storage 	 

Info: Checking placement group status
Info: 	PgState: active+clean, PgCount: 113

Info: Checking if at least one mgr pod is running
rook-ceph-mgr-a-7dc89565f5-vj879	Running	openshift-storage	ip-10-0-71-84.ec2.internal
```

File: docs/get.md [hunk 1]
```diff
@@ -12,3 +13,63 @@ This command will display the recovery-profile value set for the osd.
 $ odf get recovery-profile
 # high_recovery_ops
 ```
+
+## health
+
+Health command check health of the cluster and common configuration issues. Health command currently validates these things configurations (let us know if you would like to add other validation in health command):
```
Comment (travisn) [inline docs/get.md:19]:
```suggestion
The health command checks the health of the Ceph cluster and common configuration issues. The health command validates these configurations:
```

File: docs/get.md [hunk 2]
```diff
@@ -12,3 +13,63 @@ This command will display the recovery-profile value set for the osd.
...
+
+1. at least three mon pods should running on different nodes
+2. mon quorum and ceph health details
+3. at least three osd pods should running on different nodes
+4. all pods 'Running' status
+5. placement group status
+6. at least one mgr pod is running
+
+Health commands logs have three ways of logging:
```
Comment (travisn) [inline docs/get.md:28]: Lines 28-32 seem like they are not necessary to describe, perhaps remove them from the doc to keep it simple?

**Review by travisn — CHANGES_REQUESTED** [review]

File: docs/get.md (hunk 2 above)
Comment (subhamkrai) [reply docs/get.md]: yeah, it makes sense to remove. Done

**Comment by subhamkrai** [comment]
> based on the PR #23 , so added the label do-not-merge

JFYI @travisn

**Comment by subhamkrai** [comment]
> > based on the PR #23 , so added the label do-not-merge
> 
> JFYI @travisn

This is ready to review, I've removed the dependencies of above mentioned pr

**Comment by subhamkrai** [comment]
@travisn ^

File: cmd/odf/get/health.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	"github.com/rook/kubectl-rook-ceph/pkg/health"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/spf13/cobra"
+)
+
+var clusterHealth = &cobra.Command{
+	Use:                "health",
+	Short:              "check health of the cluster and common configuration issues",
+	DisableFlagParsing: true,
+	Hidden:             true,
```
Comment (travisn) [inline cmd/odf/get/health.go:14]: We don't want to hide this, right?

Comment (subhamkrai) [reply cmd/odf/get/health.go]: hmm we can skip this to hide as this is read-only operation I kept this Eran comment was on overall Epic. But I'll update him about this

Comment (travisn) [reply cmd/odf/get/health.go]: Yeah seems like no need to hide read only operations

**Review by travisn — APPROVED** [review]

**Pull request merged**